foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_10_0/JavaPackageMigration.java

```diff
@@ -0,0 +1,89 @@
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.baton.BatonException;
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+
+/**
+ * Updates the affected java classes from their old package name to their new package name 
+ * to ensure compatibility with the updated Java 17 dependencies 
+ */
+public class JavaPackageMigration extends AbstractAissembleMigration {
+    private static final Logger logger = LoggerFactory.getLogger(JavaPackageMigration.class);
+    private static final String[][] JAVA_PACKAGES = {
+        {"javax.activation", "jakarta.activation"},
+        {"javax.annotation", "jakarta.annotation"},
+        {"javax.batch", "jakarta.batch"},
+        {"javax.data", "jakarta.data"},
+        {"javax.decorator", "jakarta.decorator"},
+        {"javax.ejb", "jakarta.ejb"},
+        {"javax.el", "jakarta.el"},
+        {"javax.enterprise", "jakarta.enterprise"},
+        {"javax.faces", "jakarta.faces"},
+        {"javax.inject", "jakarta.inject"},
+        {"javax.interceptor", "jakarta.interceptor"},
+        {"javax.jms", "jakarta.jms"},
+        {"javax.json", "jakarta.json"},
+        {"javax.mail", "jakarta.mail"},
+        {"javax.persistence", "jakarta.persistence"},
+        {"javax.resource", "jakarta.resource"},
+        {"javax.security", "jakarta.security"},
+        {"javax.servlet", "jakarta.servlet"},
+        {"javax.transaction", "jakarta.transaction"},
+        {"javax.validation", "jakarta.validation"},
+        {"javax.websocket", "jakarta.websocket"},
+        {"javax.ws", "jakarta.ws"},
+        {"io.smallrye.reactive.messaging.providers.connectors.InMemoryConnector", "io.smallrye.reactive.messaging.memory.InMemoryConnector"},
+        {"io.smallrye.reactive.messaging.providers.connectors.InMemorySink", "io.smallrye.reactive.messaging.memory.InMemorySink"}
+    }; 
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        try {
+            String fileContent = Files.readString(file.toPath());
+
+            return Arrays.stream(JAVA_PACKAGES)
+                .map(packages -> packages[0])
+                .anyMatch(packageName -> {
+                    return fileContent.contains(packageName);
+                });
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        logger.info("Migrating file to update java class packages: {}", file.getAbsolutePath());
+        try {
+            String fileContent = Files.readString(file.toPath());
+
+            for (String[] packagePair : JAVA_PACKAGES) {
+                fileContent = fileContent.replace(packagePair[0], packagePair[1]);
```

Comment:
Q/A: We do have some sort of utility methods around replacing a string or pattern in a file. Is that functionality a good fit here? Or maybe it's too heavy handed?  Seems like we do this sort of thing enough that it would be a one liner.

Reply:
There is a baton `replaceInFile` method, but it writes the file out with each call. The way I'm currently doing it allows us to make all the necessary changes, then write out the file once at the end.